system/lib/libc/musl/src/thread/__timedwait.c

```diff
@@ -57,36 +58,34 @@ int __timedwait_cp(volatile int *addr, int val,
 		if (to.tv_sec < 0) return ETIMEDOUT;
 		top = &to;
 	}
-
 #ifdef __EMSCRIPTEN__
-	double msecsToSleep = top ? (top->tv_sec * 1000 + top->tv_nsec / 1000000.0) : INFINITY;
-	int is_runtime_thread = emscripten_is_main_runtime_thread();
+	pthread_t self = __pthread_self();
+	double msecsToSleep = top ? (top->tv_sec * 1000.0 + top->tv_nsec / 1000000.0) : INFINITY;
+	const int is_runtime_thread = emscripten_is_main_runtime_thread();
+
+	// Main runtime thread may need to run proxied calls, so sleep in very small slices to be responsive.
+	const double maxMsecsToSleep = is_runtime_thread ? 1 : 100;
```

Comment:
Split out hoisting of `self` into a separate PR?   (combine with use of `self` in 
`system/lib/libc/musl/src/thread/__wait.c`)

Reply:
Reverted with commit a9550ffb15068ee99de7b9db434b4b1f68383f8c. I'll make a new PR with this.